sci-libs/xblas/xblas-1.0.248-r2.ebuild

```diff
@@ -0,0 +1,96 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+NUMERIC_MODULE_NAME="xblas"
+FORTRAN_NEEDED=fortran
+
+inherit flag-o-matic fortran-2 toolchain-funcs out-of-source autotools
+
+DESCRIPTION="Extra Precise Basic Linear Algebra Subroutines"
+HOMEPAGE="http://www.netlib.org/xblas"
+SRC_URI="${HOMEPAGE}/${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86 ~amd64-linux ~x86-linux"
+IUSE="fortran static-libs"
+
+DEPEND=""
+RDEPEND="${RDEPEND}"
+BDEPEND="sys-devel/m4
+	fortran? ( virtual/fortran )"
+
+static_to_shared() {
+	local libstatic=${1}; shift
+	local libname=$(basename ${libstatic%.a})
+	local soname=${libname}$(get_libname $(ver_cut 1-2))
+	local libdir=$(dirname ${libstatic})
+
+	einfo "Making ${soname} from ${libstatic}"
+	if [[ ${CHOST} == *-darwin* ]] ; then
+		${LINK:-$(tc-getCC)} ${LDFLAGS}  \
+		-dynamiclib -install_name "${EPREFIX}"/usr/lib/"${soname}" \
+		-Wl,-all_load -Wl,${libstatic} \
+		"$@" -o ${libdir}/${soname} || die "${soname} failed"
+	else
+		local components=($(ver_rs 1- ' '))
+
+		${LINK:-$(tc-getCC)} ${LDFLAGS}  \
+		-shared -Wl,-soname=${soname} \
+		-Wl,--whole-archive ${libstatic} -Wl,--no-whole-archive \
```

Comment:
whole archive, followed by no whole archive...
seems kinda sus, but I'm not a linker expert.